projectq/ops/_basics.py

```diff
@@ -246,9 +258,7 @@ def matrix(self, matrix):
         self._matrix = np.matrix(matrix)
 
     def __eq__(self, other):
-        """ Return True if equal (i.e., instance of same class).
-        """
-        if not isinstance(other, self.__class__):
+        if not hasattr(other, 'matrix'):
```

Comment:
I achnowledge that the docstring was not good, but do you really want no docstring at all?

Reply:
👍